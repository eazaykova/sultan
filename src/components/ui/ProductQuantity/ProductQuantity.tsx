import { useState } from 'react';
import styles from "./productquantity.module.css";

interface ProductQuantityProps {
	amount: number;
	quantityToParent?: (quantity: number) => void;
}

const ProductQuantity = ({ amount, quantityToParent }: ProductQuantityProps) => {
	const [quantity, setQuantity] = useState(amount);

	function increment() {
		setQuantity(quantity + 1);
		quantityToParent?.(quantity + 1);
	}

	function decrement() {
		if (quantity !== 1) {
			setQuantity(quantity - 1);
			quantityToParent?.(quantity - 1);
		}
	}

	return (
		<div>
			<button className={styles.button} onClick={decrement}>-</button>
			<span className={styles.quantity}>{quantity}</span>
			<button className={styles.button} onClick={increment}>+</button>
		</div>
	)
}

export default ProductQuantity;