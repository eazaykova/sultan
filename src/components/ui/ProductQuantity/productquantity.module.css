.button {
	width: 34px;
	height: 23px;
	background: linear-gradient(90deg, rgba(255, 198, 80, 0.3) 0%, rgba(254, 202, 110, 0.3) 97.25%);
	border-radius: 1000px;

	font-weight: 300;
	font-size: 12px;
	line-height: 15px;
	text-align: center;
	color: #3F4E65;
}

.button:hover {
	background: #FFC85E;
}

.quantity {
	display: inline-flex;
	width: 15px;
	justify-content: center;

	font-weight: 500;
	font-size: 14px;
	line-height: 130%;
	color: #3F4E65;

	margin: 0 10px;
}
