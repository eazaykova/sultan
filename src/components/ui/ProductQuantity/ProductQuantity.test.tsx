import { render, screen, fireEvent } from '@testing-library/react';

import ProductQuantity from './ProductQuantity';


describe('ProductQuantity component', () => {
	test('ProductQuantity render', () => {
		render(<ProductQuantity amount={1} />);
		expect(screen.getByText(1)).toBeInTheDocument();
		expect(screen.getByText(1)).toMatchSnapshot();
	})

	test('ProductQuantity decrement', () => {
		render(<ProductQuantity amount={2} />);
		const btn = screen.getByText('-');
		fireEvent.click(btn);
		expect(screen.getByText(1)).toBeInTheDocument();
	})

	test('ProductQuantity increment', () => {
		render(<ProductQuantity amount={2} />);
		const btn = screen.getByText('+');
		fireEvent.click(btn);
		expect(screen.getByText(3)).toBeInTheDocument();
	})
})