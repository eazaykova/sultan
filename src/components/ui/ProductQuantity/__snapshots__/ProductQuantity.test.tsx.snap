// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductQuantity component ProductQuantity render 1`] = `
<span
  class="quantity"
>
  1
</span>
`;
